src/main/java/com/pinterest/secor/io/impl/MessagePackSequenceFileReaderWriterFactory.java

```diff
@@ -161,5 +175,17 @@ public void write(KeyValue keyValue) throws IOException {
         public void close() throws IOException {
             this.mWriter.close();
         }
+
+        private int numberOfFieldsMappedInHeader(KeyValue keyValue) {
+            int fields = 1;
+
+            if (keyValue.hasKafkaKey())
+                fields++;
+
+            if (keyValue.hasTimestamp())
```

Comment:
I see what you mean now, if the method only care about the number of fields, it should be fine.